src/databend/mod.rs

```diff
@@ -0,0 +1,96 @@
+use std::{borrow::Cow, collections::BTreeMap};
+
+use testcontainers::{
+    core::{wait::HttpWaitStrategy, ContainerPort, WaitFor},
+    Image,
+};
+
+const DEFAULT_IMAGE_NAME: &str = "datafuselabs/databend";
+const DEFAULT_IMAGE_TAG: &str = "v1.2.615";
+
+/// Port that the [`Databend`] container has internally
+/// Can be rebound externally via [`testcontainers::core::ImageExt::with_mapped_port`]
+///
+/// [`Databend`]: https://databend.rs/
+pub const DATABEND_PORT: ContainerPort = ContainerPort::Tcp(8000);
+
+
+/// Module to work with [`Databend`] inside of tests.
+///
+/// This module is based on the official [`Databend docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{databend, testcontainers::runners::SyncRunner};
+///
+/// let databend = databend::Databend::default().start().unwrap();
+/// let http_port = databend.get_host_port_ipv4(8000).unwrap();
+///
+/// // do something with the started databend instance.
+/// ```
+///
+/// [`Databend`]: https://databend.rs/
+/// [`Databend docker image`]: https://hub.docker.com/r/datafuselabs/databend
+#[derive(Debug, Default, Clone)]
+pub struct Databend {
+    env_vars: BTreeMap<String, String>,
+}
+
+impl Image for Databend {
+    fn name(&self) -> &str {
+        DEFAULT_IMAGE_NAME
+    }
+
+    fn tag(&self) -> &str {
+        DEFAULT_IMAGE_TAG
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::http(
+            HttpWaitStrategy::new("/").with_expected_status_code(200_u16),
+        )]
+    }
+
+    fn env_vars(
+        &self,
+    ) -> impl IntoIterator<Item = (impl Into<Cow<'_, str>>, impl Into<Cow<'_, str>>)> {
+        &self.env_vars
+    }
+
+    fn expose_ports(&self) -> &[ContainerPort] {
+        &[DATABEND_PORT]
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use databend_driver::{Client, Connection};
+    use testcontainers::ImageExt;
+    use crate::{databend::Databend as DatabendImage, testcontainers::runners::AsyncRunner};
+
+    #[tokio::test]
+    async fn test_databend() {
+        let databend = DatabendImage::default()
+            .with_env_var("QUERY_DEFAULT_USER","databend")
+            .with_env_var("QUERY_DEFAULT_PASSWORD","databend")
```

Comment:
I think we should provide some defaults and builder-like methods to override these envs (e.g `with_default_user(..)`)

WDYT?

Reply:
Because the `core/images.rs` does not provide the  `with_query_default_user` method, so we can't impl it in `impl Image for Databend`. Do you have any suggesstions?

Reply:
I meant to `impl Databend { pub fn … }`
It shouldn’t be a part of common trait.

You can check some other modules, for example: https://github.com/testcontainers/testcontainers-rs-modules-community/blob/f2bf770291d4806c1eb6aad1d0a89874b1494995/src/postgres/mod.rs#L36

Reply:
Also, you can impl Default with these env vars. And mention the default user and password in the doc comment for Databend. But user will be able to override it

Reply:
@DDtKey Thanks! I try to add the code.Plz review.